src/Stencil.php

```diff
@@ -0,0 +1,294 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Features\ZonableModelPicture;
+
+class Stencil extends CommonDBChild implements ZonableModelPicture
+{
+    public static $itemtype = "itemtype";
+    public static $items_id = "items_id";
+
+    private CommonGLPI $item;
+
+    public static function getTypeName($nb = 0): string
+    {
+        return __('Stencil');
+    }
+
+    public static function getIcon(): string
+    {
+        return 'ti ti-shape';
+    }
+
+    public static function getTable($classname = null): string
+    {
+        return 'glpi_stencils';
+    }
+
+    public function canViewStencil(): bool
+    {
+        return $this->getStencilItem()->canView();
+    }
+
+    public function canUpdateStencil(): bool
+    {
+        return $this->getStencilItem()->canUpdate();
+    }
+
+    public function canCreateStencil(): bool
+    {
+        return $this->getStencilItem()->canCreate();
+    }
+
+    public function canPurgeStencil(): bool
+    {
+        return $this->getStencilItem()->canPurge();
+    }
+
+    public static function getStencilFromItem(CommonGLPI $item): Stencil
+    {
+        $itemtype = $item::getType();
+        $items_id = $item->getID();
```

Comment:
As noted before, `getID` is not available on `CommonGLPI`.
Parameter type should probably be `CommonDBTM` instead.